### ISSUE TITLE
Reorg: hard crash with mono <=5.4.1

### ISSUE THREAD

**Issue by kohanis** [AngelDE98/MonoMod#161]
### Description
Hard crash when using mono <=5.4.1 from `PlatformDetection.OS` getter, probably something in init chain

**Comment by nike4613** [comment]
I'm unable to reproduce running under Mono provided by Unity 2017.4.40f1 (which reports as 5.0.1) under `Editor/Data/MonoBleedingEdge/bin`. That build appears to be 32-bit, however; are you seeing this in a built Unity game?

**Comment by kohanis** [comment]
Indeed, there are no problems with unity 2017, just checked. But original mono <=5.4.1 crashes. Both 32bit and 64bit, with same exit code. At least on windows
